smif/data_layer/__init__.py

```diff
@@ -1,3 +1,398 @@
+
+    def read_sector_models(self):
+        """Returns a list of excisting sector_models
+        """
+        return self._read_yaml_filenames_in_dir(self.filepath['sector_models'])
+
+    def read_sector_model(self, sector_model_name):
+        """Read a sector_model dictionary from a Yaml file
+        raises an exception when the file does not excists
+
+        Arguments
+        ---------
+        name : sector_model_name
+            String containing sector_model['name']
+        """
+        return self._read_yaml_file(self.filepath['sector_models'], sector_model_name)
+
+    def write_sector_model(self, sector_model):
+        """Write sos_model dictionary to Yaml file
+        Existing configuration will be overwritten without warning
+
+        Arguments
+        ---------
+        name : sector_model
+            Dictionary containing sector_model
+        """
+        self._write_yaml_file(self.filepath['sector_models'], sector_model['name'],
+                              sector_model)
+
+    def read_region_sets(self):
+        raise NotImplementedError()
+
+    def read_region_set_data(self, region_set_name):
+        raise NotImplementedError()
+
+    def read_interval_sets(self):
+        raise NotImplementedError()
+
+    def read_interval_set_data(self, interval_set_name):
+        raise NotImplementedError()
+
+    def read_units(self):
+        raise NotImplementedError()
+
+    def write_region_set(self, data):
+        raise NotImplementedError()
+
+    def write_interval_set(self, data):
+        raise NotImplementedError()
+
+    def write_units(self, data):
+        raise NotImplementedError()
+
+    def read_scenario_sets(self):
+        raise NotImplementedError()
+
+    def read_scenarios(self, scenario_set_name):
+        raise NotImplementedError()
+
+    def read_scenario_data(self, scenario_name):
+        raise NotImplementedError()
+
+    def write_scenario_set(self, scenario_set):
+        raise NotImplementedError()
+
+    def write_scenario(self, scenario):
+        raise NotImplementedError()
+
+    def write_scenario_data(self, scenario_name, data):
+        raise NotImplementedError()
+
+    def read_narrative_sets(self):
+        raise NotImplementedError()
+
+    def read_narratives(self, narrative_set_name):
+        raise NotImplementedError()
+
+    def read_narrative_data(self, narrative_name):
+        raise NotImplementedError()
+
+    def write_narrative_set(self, narrative_set):
+        raise NotImplementedError()
+
+    def write_narrative(self, narrative):
+        raise NotImplementedError()
+
+    def write_narrative_data(self, narrative_set_name, data):
+        raise NotImplementedError()
+
+    def _read_yaml_filenames_in_dir(self, path):
+        """Returns the name of the Yaml files in a certain directory
+
+        Arguments
+        ---------
+        path : string
+            Path to directory
+        """
+        files = list()
+        for file in os.listdir(path):
```

Comment:
The variable name `file` is reserved in python. Best to use an alternative name e.g. `filepath` or `filename`